### ISSUE TITLE
cs231n assignment1-3

### ISSUE THREAD

**Issue by JinwoongKim** [JinwoongKim/AI#18]
### softmax_loss_naive

![screen shot 2018-08-19 at 20 25 43](https://user-images.githubusercontent.com/12505517/44308332-87c24580-a3ee-11e8-9694-b02dfeda80d9.png)

```python
  loss = 0.0
  for i in xrange(num_train):
    scores = X[i].dot(W)
    score_yi = scores[y[i]]
    sum_esj = 0
    for j in xrange(num_classes):
      if j == y[i]:
        continue
      sum_esj += np.exp(scores[j])
    loss += np.log(np.exp(score_yi)/sum_esj)*-1

```
![screen shot 2018-08-19 at 20 26 12](https://user-images.githubusercontent.com/12505517/44308317-30bc7080-a3ee-11e8-854f-a58afbf93061.png)

**Comment by JinwoongKim** [comment]
Fixed bug
```python
  loss = 0.0
  for i in xrange(num_train):
    scores = X[i].dot(W)
    sum_esj = np.sum(np.exp(scores))
    loss += np.log(np.exp(scores[y[i]])/sum_esj)*-1
```

**Comment by JinwoongKim** [comment]
![screen shot 2018-08-19 at 20 36 09](https://user-images.githubusercontent.com/12505517/44308436-dbce2980-a3f0-11e8-86a5-399a27cba5c6.png)

  ```python
loss = 0.0
  for i in xrange(num_train):
    scores = X[i].dot(W)
    scores -= np.max(scores)
    sum_esj = np.sum(np.exp(scores))
    loss += np.log(np.exp(scores[y[i]])/sum_esj)*-1
```

**Comment by JinwoongKim** [comment]
For gradient descent, I found these codes,

```python
  loss = 0.0
  for i in xrange(num_train):
    scores = X[i].dot(W)
    scores -= np.max(scores)
    sum_esj = np.sum(np.exp(scores))
    loss += np.log(np.exp(scores[y[i]])/sum_esj)*-1

    for j in xrange(num_classes):
        if j == y[i]:
            dW[:, y[i]] += (-1) * (sum_esj - np.exp(scores[y[i]])) / sum_esj * X[i]
        else:
            dW[:, j] += np.exp(scores[j]) / sum_esj * X[i]
```
But couldn't found any equation and note;
